ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiParticipantPruningService.scala

```diff
@@ -44,42 +46,47 @@ final class ApiParticipantPruningService private (
   override def close(): Unit = ()
 
   override def prune(request: PruneRequest): Future[PruneResponse] = {
-    val submissionIdOrErr = Ref.SubmissionId
-      .fromString(
-        if (request.submissionId.nonEmpty) request.submissionId else UUID.randomUUID().toString
-      )
-      .left
-      .map(err =>
-        invalidArgument(s"submission_id $err")(
-          contextualizedErrorLogger(request.submissionId)
-        )
-      )
-
-    submissionIdOrErr.fold(
-      t => Future.failed(ValidationLogger.logFailure(request, t)),
-      submissionId =>
-        LoggingContext.withEnrichedLoggingContext(logging.submissionId(submissionId)) {
-          implicit loggingContext =>
-            logger.info(s"Pruning up to ${request.pruneUpTo}")
-            (for {
-
-              pruneUpTo <- validateRequest(request)(
-                loggingContext,
-                contextualizedErrorLogger(submissionId),
-              )
-
-              // If write service pruning succeeds but ledger api server index pruning fails, the user can bring the
-              // systems back in sync by reissuing the prune request at the currently specified or later offset.
-              _ <- pruneWriteService(pruneUpTo, submissionId, request.pruneAllDivulgedContracts)
-
-              pruneResponse <- pruneLedgerApiServerIndex(
-                pruneUpTo,
-                request.pruneAllDivulgedContracts,
-              )
+    val telemetryContext: TelemetryContext =
+      DefaultTelemetry.contextFromGrpcThreadLocalContext()
+    withEnrichedLoggingContext(logging.traceId(telemetryContext.traceId)) {
+      implicit loggingContext =>
```

Comment:
This 4-line pattern is repeated at least 10 times throughout the file. How about we extract it in a utils method and use that one? Aside from the repetition, I think less verbosity in these API methods would be nice as well

Reply:
replaced w/ :

https://github.com/digital-asset/daml/pull/15995/files#diff-734b593ac0044ef8aa4560732b8b405e54921f1c526421a97515cffd35e9ae51R52-R53